engine/api/src/main/java/io/deephaven/engine/table/ColumnSource.java

```diff
@@ -191,19 +192,21 @@ default <TYPE> ColumnSource<TYPE> cast(Class<? extends TYPE> clazz) {
      * retrieved from the column source, such as with {@code String myStr = colSource.get(0)}.
      *
      * @param clazz The target type.
-     * @param componentType The target component type
+     * @param componentType The target component type, may be {@code null}.
      * @param <TYPE> The target type, as a type parameter. Intended to be inferred from {@code clazz}.
      * @return A {@code ColumnSource} parameterized by {@code TYPE}.
      */
     @FinalDefault
     default <TYPE> ColumnSource<TYPE> cast(Class<? extends TYPE> clazz, Class<?> componentType) {
         final ColumnSource<TYPE> casted = cast(clazz);
-        if (!Objects.equals(componentType, casted.getComponentType())) {
-            throw new RuntimeException(String.format(
-                    "Expected componentTypes to be equal for cast with clazz=%s, componentType=%s, getComponentType()=%s",
-                    clazz, componentType, casted.getComponentType()));
+        final Class<?> columnSourceComponentType = getComponentType();
+        if ((componentType == null && columnSourceComponentType == null) || (componentType != null
+                && columnSourceComponentType != null && componentType.isAssignableFrom(columnSourceComponentType))) {
+            return casted;
         }
-        return casted;
+        throw new ClassCastException(String.format(
+                "Cannot convert column source componentType for type %s to %s (for %s)",
```

Comment:
This is prettier. Want to change the other overload to also use format? I'm a full convert to your approach, here.